ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -701,6 +699,49 @@ class JITModBuilder {
     return JITMod;
   }
 
+  void handleBranchingControlFlow(Instruction *I, size_t TraceIdx,
+                                  Function *JITFunc) {
+    // Fisrt, peek ahead in the trace and retrieve the next block. We need this
+    // so that we can insert an appropriate guard into the trace.
+    assert(InpTrace[TraceIdx + 1].hasValue()); // Should be a mappable block.
```

Comment:
I think this needs to be preceded by an assert that either: checks there is a `TraceIdx + 1`; checks that `TraceIdx` is an element which, by definition, has a successor.

Reply:
`InpTrace` has an overloaded `[]` operator that does that assertion.

We [discussed this in the last PR](https://github.com/ykjit/yk/pull/436#pullrequestreview-810106747). Same code, just moved.

Reply:
> InpTrace has an overloaded [] operator that does that assertion.

I don't think that's good enough: if this was Python I'd be asking for the same assertion, even though `[]` in Python is also memory safe. The point is that as a reader of this code I can't tell that `Idx + 1` exists from just looking at this, so I waste my limited mental capacity (very limited mental capacity :p) wondering about this -- I'm not going to know that the overloaded `[]` ensures it.

Reply:
Another block has to exist after a branch, and tracing cannot be stopped by a branch.

We can add a comment to this effect if you like.

Reply:
A brief comment would be fine!

Reply:
Try 458fae6 (which also fixes a typo).